extensions/ql-vscode/src/config.ts

```diff
@@ -449,3 +449,9 @@ export class MockGitHubApiConfigListener extends ConfigListener implements MockG
     return !!MOCK_GH_API_SERVER.getValue<boolean>();
   }
 }
+
+const MOCK_GH_API_SERVER_SCENARIOS_PATH = new Setting('mockGitHubApiServerScenariosPath', REMOTE_QUERIES_SETTING);
```

Comment:
Unfortunately, it's really hard to actually hardcode the path since we don't know where the `scenarios` directory is located. We can't really rely on the location of the extension since you can also run it outside of development mode and enable it in the production extension. In that case, we still want you to allow saving the files to the correct scenarios directory.

One thing we might be able to do is to check `ctx.extensionMode === ExtensionMode.Development` and only hardcode the path in that case (perhaps only if the path exists?).

Reply:
I think it's okay to not provide a supported path outside of development mode. We're not intending to use this functionality outside of dev. 

> One thing we might be able to do is to check ctx.extensionMode === ExtensionMode.Development and only hardcode the path in that case (perhaps only if the path exists?).

This sounds great! I wasn't aware of `extensionMode`. It could even be: try and use hard-coded path and if it doesn't exist, ask for a path to use.

Reply:
I've now made the simplest possible change which is using the hard-coded path if it exists and using the logic that was there before otherwise. We can change it to do something else when the path doesn't exist in a follow-up PR if necessary.